allensdk/brain_observatory/behavior/behavior_project_cache/project_metadata_writer/schemas.py

```diff
@@ -0,0 +1,171 @@
+import argschema
+import pathlib
+from argschema.schemas import DefaultSchema
+from marshmallow import post_load
+from marshmallow.validate import OneOf
+
+
+class BaseMetadataWriterInputSchema(argschema.ArgSchema):
+
+    behavior_nwb_dir = argschema.fields.InputDir(
+        required=True,
+        default=None,
+        allow_none=True,
+        description=(
+            "The directory where behavior_nwb sessions are "
+            "to be found when not pulling release files from LIMS. Default to "
+            "the value of ophys_nwb_dir if not set/set to None."
+        ),
+    )
+    behavior_nwb_prefix = argschema.fields.Str(
+        required=False,
+        default="behavior_session",
+        description=(
+            "Behavior session NWB files will be looked for "
+            "in the form "
+            "{behavior_nwb_dir}/"
+            "{behavior_nwb_prefix}_{behavior_session_id}.nwb"
+        ),
+    )
+
+    output_dir = argschema.fields.OutputDir(
+        required=True,
+        description=(
+            "Directory to output metadata tables and manifest."
+        ),
+    )
+
+    clobber = argschema.fields.Boolean(
+        default=False,
+        description=(
+            "If false, throw an error if output files " "already exist"
+        ),
+    )
+
+    on_missing_file = argschema.fields.Str(
+        default="error",
+        required=False,
+        validation=OneOf(("error", "warn", "skip")),
+        description=(
+            "What to do if an input datafile is missing. "
+            "If 'error', raise an exception. "
+            "If 'warn', assign a dummy ID and issue a warning. "
+            "If 'skip', do not list in metadata and issue a "
+            "warning (note, any sessions thus skipped will still "
+            "show up in aggregate metadata; there just will "
+            "be no line for those sessions in tables that list "
+            "data files for release, like sessions.csv)."
+        ),
+    )
+
+
+class BehaviorOphysMetadataInputSchema(BaseMetadataWriterInputSchema):
+
+    data_release_date = argschema.fields.List(
+        argschema.fields.String,
+        required=True,
+        cli_as_single_argument=False,
+        description=(
+            "Sub-select the set of set of sessions/experiments to release. "
+            "To set multiple, use a common separated string."
+        ),
+    )
+    ophys_nwb_dir = argschema.fields.InputDir(
+        required=True,
+        allow_none=True,
+        default=None,
+        description=(
+            "The directory where ophys exepriments are to be found when not "
```

Comment:
you modified this logic. it is now required and the description needs to be updated